### PR TITLE
Fix crash on uninferable decorators on Python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Crash identified while working on #5547 

Tagging @mbyrnepr2 since you contributed this checker and I wonder if you have an opinion on this fix.

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

File: pylint/checkers/utils.py [hunk 1]
```diff
@@ -876,7 +876,11 @@ def uninferable_final_decorators(
             except AttributeError:
                 continue
         elif isinstance(decorator, nodes.Name):
-            import_node = decorator.lookup(decorator.name)[1][0]
+            lookup_values = decorator.lookup(decorator.name)
+            if lookup_values[1]:
+                import_node = decorator.lookup(decorator.name)[1][0]
+            else:
+                continue
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/utils.py:883]: It look like this is never used, maybe we can remove it ? Beside right now nothing is exactly the same behavior, right ?

```suggestion
```

Comment (DanielNoord) [reply pylint/checkers/utils.py]: This is used on 3.6 and 3.7. Coveralls is reported on 3.8 (I believe) that's why the coverage dropped. I have added a `pragma: no cover`.

File: pylint/checkers/utils.py [hunk 2]
```diff
@@ -876,7 +876,11 @@ def uninferable_final_decorators(
             except AttributeError:
                 continue
         elif isinstance(decorator, nodes.Name):
-            import_node = decorator.lookup(decorator.name)[1][0]
+            lookup_values = decorator.lookup(decorator.name)
+            if lookup_values[1]:
+                import_node = decorator.lookup(decorator.name)[1][0]
+            else:
+                continue  # pragma: no cover # Covered on Python < 3.8
```
Comment (mbyrnepr2) [inline pylint/checkers/utils.py:883]:
```suggestion
            lookup_values = decorator.lookup(decorator.name)
            if lookup_values[1]:
                import_node = lookup_values[1][0]
            else:
                continue  # pragma: no cover # Covered on Python < 3.8
```

Comment (mbyrnepr2) [inline pylint/checkers/utils.py:883]:
```suggestion
            try:
                 import_node = decorator.lookup(decorator.name)[1][0]
             except IndexError:
                 continue
```

Comment (DanielNoord) [reply pylint/checkers/utils.py:883]: Thanks! Stupid mistake.. Glad I tagged you 😄

**Comment by mbyrnepr2** [comment]
Thank you for the headsup @DanielNoord much appreciated and thanks for catching it & fixing.
The suggestions in the code come to mind but I trust your judgement 👍

File: pylint/checkers/utils.py (hunk 2 above)
Comment (DanielNoord) [reply pylint/checkers/utils.py]: I just committed your other suggestion. Do you prefer the `try ... except`?

Comment (mbyrnepr2) [reply pylint/checkers/utils.py]: No I don't feel strongly about it; I would keep it as you have it unless you have a reason to change it.

**Review by mbyrnepr2 — APPROVED** [review]

**Pull request merged**